### ISSUE TITLE
Offer mode where app owns and initializes the SDK instance, and javaagent just instruments bytecode

### ISSUE THREAD

**Issue by johnbley** [open-telemetry/opentelemetry-java-instrumentation#3261]
**Is your feature request related to a problem? Please describe.**
The Otel Java SDK provides APIs to configure every aspect of 
telemetry collection, propagation, processing, and export.  This is great!
Since the javaagent "owns" its own private/shaded instance
of the SDK, we (the java-instrumentation project) expose a number of 
extension points to control initialization of that instance.  This 
adds additional complexity to the SDK's own config/control APIs.

**Describe the solution you'd like**
What if the -javaagent jar had a mode where it assumed that the app itself owned
the initialization of the SDK instance?  Not suggesting the real name here
but imagine `BCIInitializesSDK` as a boolean option (default `true` as in
current behavior).  If `BCIInitializesSDK=false` then the instrumentation
layer only emits bytecode inserting the usual 
`startSpan/setAttribute/addEvent/etc.` calls, and of course the necessary
context-passing instrumentation such as in the `java-concurrent` instrumentation.
Startup of the javaagent would not (in this case) initialize or own the SDK at all.

Some design issues which would need to be addressed/discussed:
- javaagent startup happens in `premain` before the app's `main` gets a chance
to initialize the SDK.  This might complicate instrumentations that keep state from
startup.
- Instrumentation of the otel api(s) would need to be disabled in the `BCIInitialziesSDK=false`
case
- We would need to document and control the version dependencies better - 
the instrumentation may require a lower API version than the SDK version 
required by the SDK-initializing-javaagent.  This might also entail some
refactoring to separate api/sdk version dependencies in the javaagent startup
sequence.
- Various extensibility features (such as configuring the private instance)
may need to be disabled or rethought for `BCIInitializesSDK=false`.
- Physical delivery - this might actually be a separate jar (without bundling the
exporters and/or SDK instance) rather than a boolean config/mode.

**Describe alternatives you've considered**
The current behavior/design is great for many use cases, including simple usage
of `@WithSpan`, normal manual instrumentation, and light customization of 
the SDK instance.  We should definitely keep it as the default!

**Additional context**
This idea has come up in a similar discussion around control/config of the .NET 
SDK instance for .NET bytecode instrumentation, where the .NET SDK hasn't grown similar extensibility concepts.  I thought it was worthwhile to bring the discussion to the Java community as well.

**Comment by jkwatson** [comment]
I can't comment on the feasibility of implementing this, but I like the idea a LOT! I would personally use this as my preferred mode of operation.

**Comment by mateuszrzeszutek** [comment]
This would work nicely for spring-sleuth/opentelemetry-spring-boot-autoconfigure users - both of those libs initialize the SDK using spring configuration. I guess that this is really convenient for spring users, they have all the config they need in one yaml/properties file, without passing long incantations of Java props via the command line. (I think we even had someone asking if we support spring config in an issue a few months ago)

The javaagent uses shaded OTel API, so a boolean switch might be impossible - all the bytecode references shaded packages and so on. But I think that it should be possible to build a separate, unshaded agent binary that does not bundle the SDK, just the instrumentations. The agent is pretty modular, and almost none of our instrumentations reference the SDK.

**Comment by johnbley** [comment]
> The javaagent uses shaded OTel API, so a boolean switch might be impossible - all the bytecode references shaded packages and so on. But I think that it should be possible to build a separate, unshaded agent binary that does not bundle the SDK, just the instrumentations. The agent is pretty modular, and almost none of our instrumentations reference the SDK.
Yes, the more I think about it the cleaner the "separate jar for unshaded pure instrumentation" idea seems.

**Comment by trask** [comment]
I agree this would be a really nice option. See https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/misc/interop-design/interop-design.md#alternate-design-1 (and the alternate design 2 that follows) for some initial challenges we'll need to think through.

**Comment by rickardoberg** [comment]
I just discovered this project and am looking into how to integrate it into our current code. This topic was one of my first questions: I need our code to own the SDK init (because single config) and still be able to use the auto instrumentation of the javaagent. If this is not possible (and this ticket has been open for a while), then effectively the javaagent and all the amazing instrumentation in it is useless, and I will have to replicate much of it myself, manually. So... any progress on this?

**Comment by trask** [comment]
hi @rickardoberg!

have you already looked into using the java agent extension mechanism to configure the SDK? https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension#extensions

**Comment by rickardoberg** [comment]
@trask In looking over those options, and also what the automated instrumentations would actually provide, in the end the best option for us is to manually configure the SDK and make it available to all other modules in our ecosystem using DI, and then they can manually create the metrics/logs/traces needed. There is too much value available here to not do a strong coupling with OT, which I think is great. Very excited about the possibilities here.
